pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -291,12 +298,31 @@ func ensureSecurityGroupRules(openstackClientFactory openstackclient.Factory, op
 		return errors.New("shootSecurityGroups must not be empty")
 	}
 
-	rules := []rules.CreateOpts{IngressAllowSSH(opt, opt.EtherType, secGroupID, opt.EtherIpAddress), EgressAllowSSHToWorker(opt, secGroupID, shootSecurityGroups[0].ID)}
-	for _, item := range rules {
-		if err := createSecurityGroupRuleIfNotExist(openstackClientFactory, item); err != nil {
-			return err
+	// remove all exist ingress allow ssh rules for patch purpose
+	ingressSSHRules, err := getRulebyName(openstackClientFactory, ingressAllowSSHResourceName(opt.BastionInstanceName))
```

Comment:
Why isn't getRulebyName scoped to one nsg? So shouldn't `SecGroupID` be set on the `ListOpts` to be sure to get only the rules for the respective security group in question, and not all?

But before fixing anything here, check my comment that starts with "Please do not just blindly delete the rule..". I strongly believe `getRulebyName` is not needed and should be deleted. Please get in touch with me if you think otherwise.

Reply:
committed in eff38149